06_puppy/n0npax/syncStore.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"errors"
+)
+
+func NewSyncStore() *SyncStore {
+	return &SyncStore{}
+}
+
+func (m *SyncStore) CreatePuppy(puppy *Puppy) int {
+	var newID int
+	m.Range(func(_, _ interface{}) bool {
```

Comment:
I also don't think it is required to iterate over the map everytime, we can turn this from a linear, to a constant runtime by storing the number of items as another variable in the `SyncStore`.

```go
type SyncStore struct {
    sync.Map
    sync.Mutex
    total    int
}
```